### PR TITLE
8283441: C2: segmentation fault in ciMethodBlocks::make_block_at(int)

### PR DESCRIPTION
**Pull request by judovana**
backport of https://github.com/openjdk/jdk/commit/947869609ce6b74d4d28f79724b823d8781adbed

Build and test passes, now running full jtregs

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8283441](https://bugs.openjdk.org/browse/JDK-8283441): C2: segmentation fault in ciMethodBlocks::make_block_at(int) (**Bug** - P3)

### Reviewers
 * [Martin Balao](https://openjdk.org/census#mbalao) (@martinuy - **Reviewer**) ⚠️ Review applies to [7fa11e40](https://git.openjdk.org/jdk8u-dev/pull/376/files/7fa11e40b713ec0c16838c21e979a2c92afe8f1a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/376/head:pull/376` \
`$ git checkout pull/376`

Update a local copy of the PR: \
`$ git checkout pull/376` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/376/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 376`

View PR using the GUI difftool: \
`$ git pr show -t 376`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/376.diff">https://git.openjdk.org/jdk8u-dev/pull/376.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/376#issuecomment-1731039924)

### CODE REVIEW

**Comment by judovana** [comment]
all jtregs seems to be passing.

**Review by martinuy — APPROVED** [review]
Hi @judovana ,

Thanks for proposing this backport. The copyright date in ciMethodBlocks.cpp could have been 2022 as in the 11u patch, but looks good to me anyways.

**Comment by judovana** [comment]
TY for review! I had changed the (C) header.

**Comment by judovana** [comment]
@martinuy  if you would be so kind to psonsor, wouldbe awesome. TY!

**Comment by martinuy** [comment]
Sure, I'll do as soon as we get 8u Maintainer approval (@gnu-andrew @jerboaa).

**Comment by judovana** [comment]
tyvm!

**Comment by gnu-andrew** [comment]
Approved for 8u.
/sponsor

**Pull request closed without merging**
